### PR TITLE
Vit5 diff clean

### PR DESCRIPTION
**Pull request by david-knigge**
This pull request introduces a new diagnostic script for diffusion model initialization and significantly updates the baseline CCNN diffusion configuration for ImageNet. The configuration changes modernize the model setup to match the JiT baseline, update patchification, kernel, and masking strategies, and adjust dataset and training parameters for improved reproducibility and performance.

**Major additions and configuration updates:**

*Model initialization diagnostics:*

- Added `benchmarks/vit5_imagenet/debug_diffusion_init.py`, a script to trace activation magnitudes layer-by-layer at initialization, helping identify where values blow up and providing detailed statistics for debugging diffusion models.

*Baseline CCNN diffusion configuration (JiT matching):*

- Renamed and relicensed the baseline config file to `examples/imagenet_diffusion/ccnn_jit_baseline.py`, updating the license to Apache-2.0 and setting copyright to NVIDIA.
- Updated dataset and training parameters: increased batch size to 256 per GPU, adjusted worker count, changed optimizer weight decay to 0, reduced training iterations to 250,000, and set EMA decay to 0.9998.
- Switched patchification modules to use `Patchify` and `Unpatchify` for in/out projections, matching JiT baseline, and set patch size dynamically from image size.
- Changed kernel in `CKConvND` from `RandomFourierKernelND` to `SIRENKernelND` with new hyperparameters, and replaced masking with `torch.nn.Identity` (no mask), simplifying and matching JiT baseline. [[1]](diffhunk://#diff-42af9006d2dc69b85b58571c579871d4da82e8fdad08e4a072ffb87d368806b9L21-R129) [[2]](diffhunk://#diff-42af9006d2dc69b85b58571c579871d4da82e8fdad08e4a072ffb87d368806b9L132-R149)
- Disabled rotary positional encoding (`use_rope=False`) to match JiT baseline.

*Removal of deprecated config:*

- Removed the old `examples/imagenet_diffusion/ccnn_12_768_hyena_qknorm.py` configuration file.

### CODE REVIEW

**Review by farhadrgh — CHANGES_REQUESTED** [review]
Pre-commit hooks are failing

**Comment by david-knigge** [comment]
> Pre-commit hooks are failing

Merged from main. I still see precommit and tests failing, but in tests/files I didn't touch, are they passing on main?

File: scripts/extract_imagenet_to_folder.py [hunk 1]
```diff
@@ -28,9 +28,9 @@
 import numpy as np
 
 
-HF_DATASET = "ILSVRC/imagenet-1k"
-HF_CACHE = os.environ.get("IMAGENET_PATH", "/shared/data/image_datasets/imagenet")
-OUTPUT_DIR = "/shared/data/image_datasets/imagenet_folder"
+HF_DATASET = os.environ.get("IMAGENET_HF_DATASET", "imagenet-1k")
+HF_CACHE = os.environ.get("IMAGENET_PATH", "/home/dknigge/project_dir/huggingface/imagenet")
```
Comment (Dafidofff) [inline scripts/extract_imagenet_to_folder.py:32]: Could you set the cluster specific path in your job scripts please :)

**Review by Dafidofff — APPROVED** [review]
LGTM, I left two co-pilot comments which are worth checking, resolved the other ones. Moreover, I ran your tests locally and they were all passing. I think you should have set the path to nvcc on our clusters before running the tests, s.t. the compile tests pass as well.

**Review by farhadrgh — APPROVED** [review]
Let's rebase, we have a CI now, no need for local pre-commit tests
